### ISSUE TITLE
Create a Table object

### ISSUE THREAD

**Issue by B-Souty** [B-Souty/html2dict#2]
While writing the docstrings, I realised it was a nightmare to describe the structure of a dictionary representing a table.

It might be easier to create a 'Table' class with an attribute for each type of table (raw html table, basic text table, rich formatted table, ...) . Or maybe even subclasses for each type. I'm not sure at this point.

I am open to suggestion.

**Comment by B-Souty** [comment]
Hey @the-c0d3r, I have completely reworked this module and it is now much simpler to use, you can have a look at it on the new [wip/issue2/main](https://github.com/B-Souty/html2dict/tree/wip/issue2/main) branch. Let me know what you think.

**Comment by the-c0d3r** [comment]
Hi @B-Souty I looked at the new branch. It looks good. I have some suggestions. Your code is a bit too long to be inside one file, maybe now's the time to create a library structure? I would put each class into a different file under `lib/`. Then use a main python file to interact with those. That will make it more organized. Until I have something solid to contribute to your repo, I can be another pair of eyes for your code review, I like doing code review on other people's code. Feel free to do the same to my repos as well. 

And I saw you used `f-strings` that's pretty cool. Didn't realize they have those until I saw it in your code and googled what is that. 


```python
if row in header_rows:
    cell_text = self.get_text_content(cell=cell, is_header=True)
else:
    cell_text = self.get_text_content(cell=cell)
```
These [4 lines](https://github.com/B-Souty/html2dict/blob/wip/issue2/main/html2dict.py#L337) of `if else` may be shortened to one line.
`cell_text = self.get_text_content(cell=cell, is_header=True if row in header_rows else False)`

**Comment by B-Souty** [comment]
@the-c0d3r Thanks for having a look, much appreciated. I'm always looking for feedback 😃. I'll check your repo as well, let me know if there is one in particular you want me to look at.

As for your suggestions:

- I will try splitting the script in multiple files as you mention, and see how it looks.
- for the `if else` I had no idea you could have ternary operator as a parameter, that's good to know 😄 . I'll give it a try but I'm afraid it will impact readability, so I might end up keeping it the more explicit way.

**Comment by B-Souty** [comment]
Fixed in #6
